internals-js/src/directiveCompositionRules.ts

```diff
@@ -0,0 +1,176 @@
+import { DirectiveLocation } from 'graphql';
+import { Directive, DirectiveDefinition, NamedSchemaElement, Schema } from './definitions'
+import { assert, assertUnreachable, isDefined } from './utils';
+
+export enum FieldPropagationStrategy {
```

Comment:
I'd rename to `FieldArgumentMergeStrategy` or the like. `Argument` because it doesn't apply to the entire field, but rather  to argument values (a field with multiple arguments will declare more than one of this). And `s/Propagation/Merge/` because this really is just a sub-part of merging/composition and we never use "propagation" for that process in other places.